### DISCUSSION TITLE
Github Actions file not running on push

### DISCUSSION THREAD

**Discussion by Bloeckchengrafik** [community/community#25690] [Actions]
<p>Hi there,<br>
my Actions file is not running on push to the <code>dev</code> branch and at this point, I just need help.<br>
It is located under <code>.github/workflows/artifact.yml</code> in the <code>master</code> branch.</p>
<pre><code class="lang-auto">name: Publish Artifact

on:
  push:
    branches: [ dev ]
  pull_request:
    branches: [ dev ]

jobs:
  build-linux:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
        with:
          ref: dev
      - name: Setup Node.js environment
        uses: actions/setup-node@v2.2.0
        with:
          node-version: 14.x
      - name: Setup npm script
        run: npm i
      - name: Build for linux
        run: npm run build-lin
      - name: Remove build files Nr. 1
        run: rm -rf dist/*-unpacked
      - name: Remove build files Nr. 2
        run: rm -rf dist/*.yml
      - name: Upload a Build Artifact
        uses: actions/upload-artifact@v2.2.4
        with:
          name: build_lin
          path: dist/*
</code></pre>
<p>thanks in advance,<br>
Christian B.</p>
<hr>
sorry for my bad english, I am still learning

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Bloeckchengrafik" data-post="1" data-topic="190074">
<div class="title">
<div class="quote-controls"></div>
 Bloeckchengrafik:</div>
<blockquote>
<p>It is located under <code>.github/workflows/artifact.yml</code> in the <code>master</code> branch.</p>
</blockquote>
</aside>
<p>For <code>push</code> events, the workflow file must be on the branch being pushed. Try committing your workflow to the <code>dev</code> branch.</p>

**Comment by Bloeckchengrafik** [comment]
<p>Thanks a lot! That solves the problem!</p>
